ref/general/microshed.adoc

```diff
@@ -6,78 +6,87 @@
+Integration tests assess multiple test classes and components at once, and are better at analyzing the working of the different parts of your application as a whole.
+Therefore, developers find integration testing to be more reliable for their application to work in production environment.
+However, integration tests take longer and can be complicated to set up and configure.
+With faster development cycles and limited time available to developers, it's not always feasible to run integrated tests for your applications.
 
-== What challenges does MicroShed resolve
+== Development-production parity
 
-An application might run differently in production as compared to the development and testing environment.
+Development-production parity, or dev-prod parity, is one of the factors in the link:https://12factor.net/[twelve-factor app], a methodology to build modern applications.
+The idea behind dev-prod parity is to keep the development, staging, and production environments as similar as possible, and it addresses gaps in development ad production in terms of time, personnel, and tools.
+Developers mostly use different tools in development as compared to production for speed and ease of use.
+For example you might use a local maven build in development, but the application might be deployed to a docker container in production.
 Differences between the environments can cause a test to fail in production, even though it passed in development.
-You can address this issue by testing your application in the same container that you use in production.
-
-Running tests in development mode is convenient for local development, but it can be tedious to test against a running Open Liberty server in production scenarios, such as CI/CD pipelines.
-For this reason, MicroShed Testing can start and stop the application runtime before and after the tests are run.
-
-== How it works
+The application might work on your machine but not in production.
+So it is important to use the same services in development as in production.
 
-You can use MicroShed Testing to develop integration tests for your Open Liberty application.
-You test your application from outside its container so that the tests run against the same image that you use in production.
+== Using containers 
 
-To write a test with MircoShed testing, first add `microshed-testing-testcontainers` and `junit-jupiter` artifact IDs as test-scoped dependencies, as shown in the following example:
-
-[source,xml]
-----
-<dependency>
-    <groupId>org.microshed</groupId>
-    <artifactId>microshed-testing-testcontainers</artifactId>
-    <version>0.4.1</version>
-    <scope>test</scope>
-</dependency>
-
-<!-- Any compatible version of JUnit Jupiter 5.X will work -->
-<dependency>
-    <groupId>org.junit.jupiter</groupId>
-    <artifactId>junit-jupiter</artifactId>
-    <version>5.4.2</version>
-    <scope>test</scope>
-</dependency>
-----
+The programming paradigms have changed with how we test software.
```

Comment:
New programming paradigms are changing how we test software.

is that what you mean? As written, it sounds like the tests changed the paradigm

Reply:
It kind of means that paradigms have changed about how we test software because there it is no more needed to use different test tools in  development and production. But I took that sentence away.